docs/Reboots.md

```diff
@@ -0,0 +1,29 @@
+# What causes reboots?
+
+Under certain conditions, MachineConfig updates will cause the Machine
+Config Daemon to reboot the nodes affected by the change. This document
+aims to identify exactly what MachineConfig changes cause a reboot, with
+the intent that any changes will not be unexpected.
+
+A MachineConfig change will cause a reboot under the following
+conditions:
+
+-   An OS update is being applied.
+-   Linux kernel arguments have changed.
```

Comment:
Make this list shorter and link to the explicit terms: for example KernelType with link to https://github.com/openshift/machine-config-operator/blob/ca8fcb887f153604556f2706cee8a33165879797/docs/MachineConfiguration.md#kerneltype

Reply:
Great idea! I'll do that for each item.